delegator.py

```diff
@@ -182,7 +182,11 @@ def run(self, block=True, binary=False, cwd=None, env=None):
                 popen_kwargs['cwd'] = cwd
             if env:
                 popen_kwargs['env'].update(env)
-            s = subprocess.Popen(self._popen_args, **popen_kwargs)
+            args = self._popen_args
+            if not isinstance(args, str):
```

Comment:
Need to account for `unicode` string type in PY2.

Reply:
If python2 is still relevant, are you fine with introducing `six` as a new dependency and use `six.string_types`?